src/services/DfusionService.ts

```diff
@@ -146,7 +146,7 @@ export class DfusionRepoImpl implements DfusionService {
 
   public watchOrderPlacement(params: WatchOrderPlacementParams) {
     this._contract.events
-      .OrderPlacement()
+      .OrderPlacement({ fromBlock: 6348195 }) // allows to get a bunch of past orders at once
```

Comment:
This made me think. If we can keep `fromBlock` between bot restarts on updates or failure, we could easily resume from exact block of shutdown

Reply:
This is for testing only, will remove later

Reply:
Like, on every new block successfully parsed, update a local file `lastBlock.txt` or something, and on start try to look for that file?

Reply:
Yes, there's the idea of persisting the last processed block (you update when you successfully notified all the orders from that block)

One simple way of persisting that is good enough here, is using a sqllite database

This would require also to change the events to be based on get past logs. We would be much more reliable if we take those steps

Reply:
No, no `.OrderPlacement({ fromBlock: 6348195 })` allows us to get all events starting from block, past + new one, no need for `getPastEvents` then

Reply:
Yes, but getPastEvents was to choose discrete blocks instead of "watching":

* listen for new block
* handle notifications of block
* persist the last handled block

Reply:
Unless we need to differentiate between past and ongoing events, does it matter?
`.OrderPlacement({ fromBlock: 6348195 }).on('data'` first gets triggered for every event starting `fromBlock` and then for every new event

It's kinda like
```
.getPastEvents({ fromBlock: 6348195 }) + .OrderPlacement()
```

Reply:
I ran into another issue just today.
When there are too many events in a row all the blockchain calls in `.on('data'`
```
const [sellTokenAddress, buyTokenAddress] = await Promise.all([
          this._getTokenAddress(sellTokenId),
          this._getTokenAddress(buyTokenId),
        ])
        const [sellToken, buyToken, validFrom, validUntil] = await Promise.all([
          this._getToken(sellTokenAddress),
          this._getToken(buyTokenAddress),
          this._batchIdToDate(validFromBatchId),
          this._batchIdToDate(validUntilBatchId.plus(1)), // adding 1 since validUntillBatchId is inclusive. https://github.com/gnosis/dex-telegram/issues/183
        ])
```

get stuck in limbo and never resolve
need to cache and/or throttle